vm/src/pyobject.rs

```diff
@@ -753,7 +776,13 @@ where
     type Target = T;
 
     fn deref(&self) -> &T {
-        self.obj.payload().expect("unexpected payload for type")
+        self.obj.payload().unwrap_or_else(|| {
+            if cfg!(debug_assertions) {
+                panic!("PyRef.obj did not have a payload of T")
+            }
```

Comment:
I think `unreachable_unchecked()` still panic for opt 0. But to force check it in debug mode, this is also possible:
```
debug_assert!(false, "PyRef.obj did not have a payload of T");
```